Build failure: rocmPackages.llvm.compiler-rt-6.0.2

### Steps To Reproduce

Steps to reproduce the behavior:
1. build rocmPackages.llvm.compiler-rt

### Build log

```bash
@nix { "action": "setPhase", "phase": "checkPhase" }
Running phase: checkPhase
check flags: -j24 check-runtimes
[0/1] Running runtimes regression tests
/build/source/compiler-rt/test/lit.common.cfg.py:639: SyntaxWarning: invalid escape sequence '\$'
  config.substitutions.append( ("%ld_flags_rpath_exe" + postfix, "-Wl,-z,origin -Wl,-rpath,\$ORIGIN -L%T -l%xdynamiclib_namespec" + postfix) )
/build/source/compiler-rt/test/lit.common.cfg.py:642: SyntaxWarning: invalid escape sequence '\$'
  config.substitutions.append( ("%ld_flags_rpath_exe" + postfix, "-Wl,-rpath,\$ORIGIN -L%T -l%xdynamiclib_namespec" + postfix) )
/build/source/compiler-rt/test/lit.common.cfg.py:645: SyntaxWarning: invalid escape sequence '\$'
  config.substitutions.append( ("%ld_flags_rpath_exe" + postfix, "-Wl,-R\$ORIGIN -L%T -l%xdynamiclib_namespec" + postfix) )
lit: /build/source/compiler-rt/test/lit.common.cfg.py:148: warning: Compiler lib dir != compiler-rt lib dir
Compiler libdir:     "/nix/store/kjav83vr3vhq96hj4wr95fbkxdgwcbqg-rocm-llvm-runtimes-6.0.2/lib/linux"
compiler-rt libdir:  "/build/source/runtimes/build/compiler-rt/lib/linux"
lit: /build/source/compiler-rt/test/lit.common.cfg.py:158: warning: COMPILER_RT_TEST_STANDALONE_BUILD_LIBS=ON, but this test suite does not support testing the just-built runtime libraries when the test compiler is configured to use different runtime libraries. Either modify this test suite to support this test configuration, or set COMPILER_RT_TEST_STANDALONE_BUILD_LIBS=OFF to test the runtime libraries included in the compiler instead.
lit: /build/source/compiler-rt/test/lit.common.cfg.py:169: note: Testing using libraries in "/build/source/runtimes/build/compiler-rt/lib/linux"
lit: /nix/store/yz7yczcfpipg2gdbs5xm9pxxl84h1pq7-python3.12-lit-18.1.7/lib/python3.12/site-packages/lit/TestingConfig.py:152: fatal: unable to parse config file '/build/source/compiler-rt/test/lit.common.cfg.py', traceback: Traceback (most recent call last):
  File "/nix/store/yz7yczcfpipg2gdbs5xm9pxxl84h1pq7-python3.12-lit-18.1.7/lib/python3.12/site-packages/lit/TestingConfig.py", line 140, in load_from_path
    exec(compile(data, path, "exec"), cfg_globals, None)
  File "/build/source/compiler-rt/test/lit.common.cfg.py", line 519, in <module>
    from distutils.version import LooseVersion
ModuleNotFoundError: No module named 'distutils'

FAILED: CMakeFiles/check-runtimes /build/source/runtimes/build/CMakeFiles/check-runtimes
cd /build/source/runtimes/build && /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12 /nix/store/yz7yczcfpipg2gdbs5xm9pxxl84h1pq7-python3.12-lit-18.1.7/bin/.lit-wrapped -sv --show-xfail --show-unsupported /build/source/runtimes/build/compiler-rt/test/builtins/TestCases /build/source/runtimes/build/compiler-rt/test/builtins/Unit/X86_64LinuxConfig
ninja: build stopped: subcommand failed.
```

### Additional context

distutils module was removed in python 3.12

### Notify maintainers
@acowley @lovesegfault

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/rocm-modules/6/llvm/base.nix
+++ b/pkgs/development/rocm-modules/6/llvm/base.nix
@@ -86,7 +86,7 @@ in stdenv.mkDerivation (finalAttrs: {
     cmake
     ninja
     git
-    python3Packages.python
+    (python3Packages.python.withPackages (p: [ p.setuptools ]))
   ] ++ lib.optionals (buildDocs || buildMan) [
     doxygen
     sphinx

--- a/pkgs/development/rocm-modules/6/llvm/stage-2/1000-libcxx-failing-tests.list
+++ b/pkgs/development/rocm-modules/6/llvm/stage-2/1000-libcxx-failing-tests.list
@@ -169,3 +169,7 @@
 ../libcxx/test/std/utilities/utility/mem.res/mem.res.pool/mem.res.pool.mem/unsync_allocate_overaligned_request.pass.cpp
 ../libcxx/test/std/utilities/utility/mem.res/mem.res.pool/mem.res.pool.mem/unsync_allocate_reuse_blocks.pass.cpp
 ../libcxx/test/std/utilities/utility/mem.res/mem.res.pool/mem.res.pool.mem/unsync_allocate.pass.cpp
+../libcxx/test/std/language.support/support.dynamic/hardware_inference_size.compile.pass.cpp
+../libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp
+../libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.single/sized_delete14.pass.cpp
+../libcxx/test/libcxx/selftest/sh.cpp/empty.sh.cpp

